lib/signs_ui/signs/expiration.ex

```diff
@@ -0,0 +1,78 @@
+defmodule SignsUI.Signs.Expiration do
+  @moduledoc """
+  Handle cleaning up settings in the configuration that have since expired
+  """
+  use GenServer
+  alias SignsUI.Signs.Sign
+
+  @type state :: %{
+          time_fetcher: (() -> DateTime.t()),
+          loop_ms: integer(),
+          sign_state_server: GenServer.server()
+        }
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(opts \\ []) do
+    name = Keyword.get(opts, :name, __MODULE__)
+
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  @spec init(Keyword.t()) :: {:ok, state()}
+  def init(opts) do
+    time_fetcher = Keyword.get(opts, :time_fetcher, fn -> DateTime.utc_now() end)
+    loop_ms = Keyword.get(opts, :loop_ms, 5_000)
+    sign_state_server = Keyword.get(opts, :sign_state_server, SignsUI.Signs.State)
+    schedule_loop(self(), loop_ms)
+    {:ok, %{time_fetcher: time_fetcher, loop_ms: loop_ms, sign_state_server: sign_state_server}}
+  end
+
+  @spec handle_info(:process_expired, state()) :: state()
+  def handle_info(:process_expired, state) do
+    updates =
+      state.sign_state_server
+      |> SignsUI.Signs.State.get_all()
+      |> expire_signs(state.time_fetcher)
+
+    if updates != %{} do
+      SignsUI.Signs.State.update_some(state.sign_state_server, updates)
```

Comment:
Maybe a `Logger.info` here, so we can more easily see if it's happening (or not)?